File: ObjC/ASTM30View-ObjC/ASTM30View-ObjC/ASTM30GraphicViewController/ASTM30PointsInfo.m
//
//  ASTM30PointsInfo.m
//  ASTM30View-ObjC
//
//  Created by Inaba Mizuki on 2016/1/26.
//  Copyright © 2016年 Inaba Mizuki. All rights reserved.
//

#import "ASTM30PointsInfo.h"

@implementation ASTM30Point

@synthesize key;
@synthesize value;

- (instancetype)initWithKey:(NSString *)aKey value:(CGPoint)aValue {
    self = [super init];
    if (self == nil) return self;

    self.key = aKey;
    self.value = aValue;


    return self;
}

@end



@implementation ASTM30PointsInfo

@synthesize name;
@synthesize lineWidth;
@synthesize color;
@synthesize colorInMasked;
@synthesize points;
@synthesize closePath;

- (instancetype _Nonnull)initWithName:(NSString * _Nonnull)aName {
    self = [super init];
    if (self == nil) return self;

    self.name = name;

    return self;
}

- (ASTM30Point * _Nullable)pointWithKey:(NSString * _Nonnull)key {
    for (ASTM30Point* point in points) {
        if ([point.key isEqualToString:key]) {
            return point;
        }
    }

    return nil;
}

@end

File: Swift/ASTM30View-Swift/ASColorVectorViewProj/ASTM30GraphicViewController/ASTM30PointsInfo.swift
//
//  ASTM30Point.swift
//
//  Created by Inaba Mizuki on 2016/1/22.
//  Copyright © 2016年 AsenseTek. All rights reserved.
//

import UIKit

// MARK: ASTM30Point

class ASTM30Point: NSObject {

    var key: String

    var value: CGPoint

    init(key: String, value: CGPoint) {
        self.key = key
        self.value = value
        
        super.init()
    }
}


// MARK: ASTM30PointsInfo

class ASTM30PointsInfo: NSObject {

    var name: String

    var lineWidth: CGFloat = 2.0

    var color = UIColor.blackColor()

    var colorInMasked: UIColor? = nil

    var points = [ASTM30Point]()

    var closePath: Bool = true

    init(name: String) {
        self.name = name

        super.init()
    }

    func pointWithKey(key: String) -> ASTM30Point? {
        return points.filter { point in point.key == key }.first
    }
}
